wicker/core/datasets.py

```diff
@@ -22,6 +26,81 @@
 FILE_LOCK_TIMEOUT_SECONDS = 300
 
 
+def get_file_size_s3_optional_copy_gcloud(input_tuple: Tuple[List[Tuple[str, str]], ValueProxy, Lock, bool, str, str]):
```

Comment:
Sorry I am not following why a `get` function should do heavy weight copy operations?

Reply:
Currently Wicker already parses through the col files and has the names and uris, it is imo the most logical place to put this code as the parsing has already been completed. These process both have to use the file path search. The get is just reading the byte size through the s3 client.